HIP/hip-0000-event-serialization.md

```diff
@@ -0,0 +1,227 @@
+---
+hip: <HIP number (this is determined by the HIP editor)>
+title: Event Serialization With Round Number, EventDescriptors, and Multiple Other Parents.
+author: Edward Wertz <edward@swirldslabs.com>
+working-group: Swirlds Labs Platform Hashgraph Team
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Draft
+created: 2023-10-25
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/827
+updated: <comma separated list of dates>
+requires: <HIP number(s)>
+replaces: <HIP number(s)>
+superseded-by: <HIP number(s)>
+---
+
+## Abstract
+
+This HIP proposes several changes to the event serialization format.
+
+1. Add the round during which an event was created to the event's meta-data.
+2. Represent links to event parents as EventDescriptors, 4-tuples of (Event Hash, Creator NodeId, Generation, Round
+   Number).
+3. Allow for multiple other parents.
+4. Reduce an event's unhashed meta-data to just the event's signature.
+
+## Motivation
+
+### Add Address Book Round Number
+
+When we allow dynamic address books such that nodes can be added or removed without restart, we need a way to indicate
+which address book was active when an event was created. A new address book becomes active on a round boundary. Writing
+the current round number into an event's meta-data upon creation will provide a way to determine which
+address book to use when validating the event.
+
+This round number will also be used to determine when an event has become ancient and is no longer able to come to
+consensus if it has not reached consensus yet.
+
+### Represent Links to Event Parents as EventDescriptors
+
+An EventDescriptor is a 4-tuple:
+
+1. Event Hash - The hash of the event.
+2. Creator NodeId - The NodeId of the node that created the event.
+3. Generation - The generation of the event. (1 + the max of the event's parent's generations)
+4. Round Number - The round during which the event was created.
+
+The EventDescriptor is the minimal amount of information needed to identify an event by its hash, which node created
+it, and be able to determine when an event has become ancient. It is used in place of the event itself when the
+content of the event (it's transactions) are not needed. This encapsulation of meta-data will allow us to further
+improve gossip, validation, and consensus algorithms.
+
+### Allow for Multiple Other Parents
+
+An event's ability to come to consensus is based on whether witness events can strongly see the
+event through parent links. Allowing for multiple other parents will improve the ability of events to be
```

Comment:
Nit: "allowing for" sounds like a synonym for "accounting for", which isn't the intended reading.

Maybe "Permitting multiple other parents..." is a clearer expression?

Reply:
removed allowing for since this is a title of a section.